### DISCUSSION TITLE
composite of two images results in unexpected 90º rotation of second image

### DISCUSSION THREAD

**Discussion by joeldcanfield** [ImageMagick/ImageMagick#7587] [Digital Image Processing] [answered]
this command using the first two attached images results in the third image. how is the rotation of gotg021.jpg happening?

`magick composite -blend  50 gotg021.jpg gotg01150.jpg -alpha Set transition1-2-50.jpg;`

![gotg021](https://github.com/user-attachments/assets/9d8360f5-0806-4d7a-ba41-9a772aae9483)
![gotg01150](https://github.com/user-attachments/assets/94fa95d6-b312-4761-b707-c47d40308f44)
![transition1-2-50](https://github.com/user-attachments/assets/cfcfed3b-3502-4829-89dc-0fbebbe5a6d8)

**Comment by dceeee** [comment]
I named the files first.jpg & second.jpg
magick first.jpg -format %[orientation] info:
TopLeft
magick second.jpg -format %[orientation] info:
RightTop

I then did
magick second.jpg -auto-orient secondao.jpg
magick composite -blend 50 first.jpg secondao.jpg blendout.jpg
which seems to work, I'm pretty sure that the -alpha set does nothing
Hopefully someone can provide a one line command

**Reply by joeldcanfield** [reply]
auto-orient seems to be the issue; thanks

**Comment by snibgo** [comment, marked as answer]
For that photo (01150), the camera was rotated by 90 degrees. This fact is recorded in metadata:
```
C:\web\im>%IMG7%magick gotg021.jpg -verbose info: |grep Orientation
  Orientation: TopLeft

C:\web\im>%IMG7%magick gotg01150.jpg -verbose info: |grep Orientation
  Orientation: RightTop
```
Some software will read this metadata and automatically rotate the image to compensate. IM won't do this unless you tell it to, with `-auto-orient`. That option isn't available in `magick compose`. Instead, use just `magick`, with the input images in the opposite order, like this:
```
magick gotg01150.jpg gotg021.jpg -auto-orient -compose Blend -define compose:args=50 -composite out.jpg
```
In your example, `-alpha set` has no effect.

**Reply by joeldcanfield** [reply]
thanks so much for the clear explanation. like learning HTML 30 years ago and CSS 20 years ago, I'm still at the "view source/copy/paste/muck about" stage with most Imagemagick commands, so artifacts like the `-alpha set` from the example using pngs just got carried over, and obviously I was clueless about the lack of/need for orientation in the command.
